src/source.rs

```diff
@@ -19,21 +19,37 @@ use std::collections::HashMap;
 use std::fmt::{Display, Formatter};
 use std::fs::OpenOptions;
 use std::io::{Read, Seek, SeekFrom};
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use std::time::Duration;
 
+use crate::blockdev::*;
 use crate::cmdline::*;
 use crate::errors::*;
+use crate::osmet::*;
 
 /// Completion timeout for HTTP requests (4 hours).
 const HTTP_COMPLETION_TIMEOUT: Duration = Duration::from_secs(4 * 60 * 60);
 
 /// Default base URL to Fedora CoreOS streams metadata.
 const DEFAULT_STREAM_BASE_URL: &str = "https://builds.coreos.fedoraproject.org/streams/";
 
+/// Marker file for whether we're in a live OSTree system.
+const LIVE_SYSTEM_MARKER: &str = "/run/ostree-live";
+
+/// Marker file for whether we're in a CoreOS system.
+const COREOS_SYSTEM_MARKER: &str = "/sysroot/.coreos-aleph-version.json";
+
+/// Directory in which we look for osmet files.
+const OSMET_FILES_DIR: &str = "/run/coreos-installer/osmet";
```

Comment:
`/run` implies that some external process needs to put them there.  Why not `/usr/share`?

Reply:
We copy it into `/run` from the initrd. It can't be part of `/usr` since that'd require making it part of the metal image we've just packed. :)  See https://github.com/coreos/fedora-coreos-config/pull/322.